sys/stdvga.c

```diff
@@ -88,6 +91,45 @@ static int stdvga_palette_write(vga_device_t *vga, uio_t *uio) {
   return 0;
 }
 
+static int stdvga_get_resolution(vga_device_t *vga, uint16_t *xres,
+                                 uint16_t *yres, unsigned int *bpp) {
+  stdvga_device_t *stdvga = STDVGA_FROM_VGA(vga);
+  *xres = stdvga->width;
+  *yres = stdvga->height;
+  *bpp = stdvga->bpp;
+  return 0;
+}
+
+static int stdvga_set_resolution(vga_device_t *vga, uint16_t xres,
+                                 uint16_t yres, unsigned int bpp) {
+  stdvga_device_t *stdvga = STDVGA_FROM_VGA(vga);
+
+  /* I have not tested larger resolutions yet. */
+  if (xres > 640 || yres > 480)
```

Comment:
Actually, BochsVBE extensions are very versatile, and apparently QEMU's stdvga accepts arbitrary dimensions for screen resolution - it would work correctly even with weird stuff like `12345 x 1`. I've only added this condition to impose some reasonable limit on input numbers.

This will be very different for `cirrus` driver for OVPsim (which I've also started working on), as the configuration for each video mode will be very different. Thus I intend to support only some selected video modes, and indeed checking against a list will be the only feasible solution.